ios/RNRadar.m

```diff
@@ -179,44 +179,25 @@ - (void)didLogMessage:(NSString *)message {
     }];
 }
 
-RCT_EXPORT_METHOD(trackOnce:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject) {
+RCT_EXPORT_METHOD(trackOnce:(NSDictionary *)optionsDict resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject) {
+    RadarTrackingOptionsDesiredAccuracy desiredAccuracy;
+    BOOL beaconsTrackingOption = NO;
+    desiredAccuracy = RadarTrackingOptionsDesiredAccuracyMedium;
+
     __block RCTPromiseResolveBlock resolver = resolve;
     __block RCTPromiseRejectBlock rejecter = reject;
 
-    [Radar trackOnceWithCompletionHandler:^(RadarStatus status, CLLocation * _Nullable location, NSArray<RadarEvent *> * _Nullable events, RadarUser * _Nullable user) {
-        if (status == RadarStatusSuccess && resolver) {
-            NSMutableDictionary *dict = [NSMutableDictionary new];
-            [dict setObject:[Radar stringForStatus:status] forKey:@"status"];
-            if (location) {
-                [dict setObject:[Radar dictionaryForLocation:location] forKey:@"location"];
-            }
-            if (events) {
-                [dict setObject:[RadarEvent arrayForEvents:events] forKey:@"events"];
-            }
-            if (user) {
-                [dict setObject:[user dictionaryValue] forKey:@"user"];
-            }
-            resolver(dict);
-        } else if (rejecter) {
-            rejecter([Radar stringForStatus:status], [Radar stringForStatus:status], nil);
-        }
-        resolver = nil;
-        rejecter = nil;
-    }];
-}
+    NSDictionary *locationDict = optionsDict[@"location"];
 
-RCT_EXPORT_METHOD(trackOnce:(NSDictionary *)locationDict resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject) {
```

Comment:
So one thing I'm realizing @brettguenther is that the old code here supported this format:
```
Radar.trackOnce({
  latitude: 39.2904,
  longitude: -76.6122,
})
```
And the new code supports nesting `latitude` and `longitude` under `location { ... }` like this:
```
Radar.trackOnce({
  location: {
    latitude: 39.2904,
    longitude: -76.6122,
  }
})
```

So this change would probably break any existing callers.

Wondering if we should keep the old format for backcompat

cc @gabrielalao

Reply:
@tjulien We can make our code back compatible. I can update javascript code for handling parameters.